src/main/java/frc/utils/calibration/sysid/SysIdCalibrator.java

```diff
@@ -0,0 +1,130 @@
+import frc.utils.GBSubsystem;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.function.Consumer;
+
+import static edu.wpi.first.units.Units.Seconds;
+import static edu.wpi.first.units.Units.Volts;
+
+public class SysIdCalibrator {
+
+    private final SysIdRoutine sysIdRoutine;
+
+    private final GBSubsystem usedSubSystem;
+
+    private final boolean isCTRE;
+
+    /**
+     * IMPORTANT:
+     *
+     * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
+     */
+    public SysIdCalibrator(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl, double voltageStepVolts,
+                           double rampRateVoltsPerSecond) {
+        this(
+                isCTRE,
+                subsystem,
+                voltageSetControl,
+                voltageStepVolts,
+                rampRateVoltsPerSecond,
+                SysIdConstants.DEFAULT_TIMEOUT_SECONDS
+        );
+    }
+
+    /**
+     * IMPORTANT:
+     *
+     * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
+     */
+    public SysIdCalibrator(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl, double voltageStep) {
+        this(
+                isCTRE,
+                subsystem,
+                voltageSetControl,
+                voltageStep,
+                SysIdConstants.DEFAULT_RAMP_RATE_VOLTS_PER_SECOND,
+                SysIdConstants.DEFAULT_TIMEOUT_SECONDS
+        );
+    }
+
+    /**
+     * IMPORTANT:
+     *
+     * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
+     */
+    public SysIdCalibrator(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl) {
+        this(
+                isCTRE,
+                subsystem,
+                voltageSetControl,
+                SysIdConstants.DEFAULT_VOLTAGE_STEP,
+                SysIdConstants.DEFAULT_RAMP_RATE_VOLTS_PER_SECOND,
+                SysIdConstants.DEFAULT_TIMEOUT_SECONDS
+        );
+    }
+
+    /**
+     * IMPORTANT:
+     *
+     * @param voltageSetControl - note that this function needs to use kg in it so the mechanism won't move because of gravity.
+     */
+    public SysIdCalibrator(boolean isCTRE, GBSubsystem subsystem, Consumer<Double> voltageSetControl, double voltageStepVolts,
+                           double rampRateVoltsPerSecond, double timeoutSeconds) {
+        this.usedSubSystem = subsystem;
+        this.isCTRE = isCTRE;
+
+        final SysIdRoutine.Config config = new SysIdRoutine.Config(
+                Volts.of(rampRateVoltsPerSecond).per(Seconds.of(1)),
+                Volts.of(voltageStepVolts),
+                Seconds.of(timeoutSeconds),
+                this.isCTRE
+                        ? (state) -> SignalLogger.writeString("state", state.toString())
```

Comment:
```suggestion
                this.isCTRE ?
                        (state) -> SignalLogger.writeString("state", state.toString())
```
i prefer it like this